### ISSUE TITLE
Bug: `max_steps` is missing in learning rate annealing schedulers, causing incorrect decay

### ISSUE THREAD

**Issue by XChen-Zero** [NVIDIA/NeMo#14186]
In the `[adam.py](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/recipes/optim/adam.py)` optimizer configuration, the learning rate schedulers func that implement annealing are missing the crucial `max_steps` parameter.

Without `max_steps`, the scheduler does not know the total duration of the training and cannot compute the learning rate decay correctly. This leads to the learning rate dropping directly to `min_lr` at the beginning of the training run, instead of gradually annealing over the course of training.

case:

https://github.com/NVIDIA/NeMo/blob/d5cfc1185aa696477ee368a8cf5adf2471f4fdc7/nemo/collections/llm/recipes/optim/adam.py#L67-L97

This function needs to pass max_step to the CosineAnnealingScheduler!!!

**Comment by maanug-nv** [comment]
Hi thanks for bringing this up.

The MegatronStrategy is currently copying over the Trainer's max_steps to the lr scheduler [here](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/strategies/megatron_strategy.py#L464-L465). Currently, it didn't seem necessary to push this responsibility to the user, so we didn't expose it in the optimizer recipe.

Are you still seeing training runs with incorrect lr decay despite those lines? I suppose if you are using that optimizer recipe outside the MegatronStrategy, you'd run into that issue.

**Comment by maanug-nv** [comment]
Hi @XChen-Zero , do you still see incorrect lr decay despite the lines mentioned above?
